Cargo.toml

```diff
@@ -41,7 +41,7 @@ byteorder = "1.3.1"
 env_logger = "0.7.1"
 
 [features]
-default = []
+default = ["rust-crypto"]
```

Comment:
Is this a left-over from debugging, or should `rust-crypto` really be a default reagure?

Reply:
This is because we need to take SHA-256 from some "crypto-provider" crate in order to implement Sha256Hasher. Previously we didn't need it in default feature since the only hasher was XOR128 and it didn't use any crypto under the hood. I decided to take `rust-crypto`, because it looks "more transparent" then `ring` one, which actually involves `boringssl` C library.

Reply:
`rust-crypto` seems to be used only by tests and benchmarks. So I propose making it a non-optional dev dependency instead. This way it won't be part of the dependency tree if you have `merkletree` as a dependency.

Reply:
It seems that we can't do that without significant reorganisation of the codebase actually. If we make `rust-crypto` a dev-dependency, than we will need to move (and start thinking about them as integration tests, which they are by their nature) tests to /tests directory from the following list of files:
- test_cmh.rs
- test_common.rs
- test_item.rs
- test_sip.rs
- test_xor128.rs

which "historically" was considered as unit-tests and was actually a part of merkletree source code (located in `/src` directory under `#cfg(test)` macro).

But in general I agree that this change should be introduced because source code of this repository is not an appropriate place for actual hasher's implementations. E.g. in merkletree repository we provide hasher's abstractions and interfaces but implementing hasher is considered to be a task for external caller (like [rust-file-proofs](https://github.com/filecoin-project/rust-fil-proofs/tree/75d6fd8b96cfa85821b56a8bf59b88882d446015/filecoin-hashers/src) caller, where we have Blake2s, Poseidon and Sha256 hashers). 

This seems to be a rather big change which I would make as a next separate incremental PR. WDYT?

Reply:
Separate PR sounds fine, thanks

Reply:
I agree that a separate PR for re-organizing the tests makes sense and also that this re-organizing should happen. Though I don't think it's needed. As long as the tests are behind `#![cfg(test)]`, it should work having `rust-crypto` as dev dependency.

Reply:
Yes exactly! See my other comment, that you need to use `#![cfg(test)]` so that it is applied to the whole file.